### PR TITLE
Feat modal

### PR DESCRIPTION
**Pull request by junaidshaikh-js**
Added modal example and live demo of modal

Preview: https://deploy-preview-36--hydrogen-ui.netlify.app/docs/modal.html

### CODE REVIEW

**Review by pratik-kumar-sahu — APPROVED** [review]
- Looks good to be merged. Go ahead

File: docs/css/docs.css [hunk 1]
```diff
@@ -120,6 +120,33 @@
   margin-bottom: 0 !important;
 }
 
+/* example modal */
+#example-modal {
+  background-color: rgba(0, 0, 0, 0.8);
```
Comment (devananth) [inline docs/css/docs.css:125]: `css variables`  can be used

**Review by devananth — CHANGES_REQUESTED** [review]
Everything is fine , Good work !

**Review by lannisterrr — COMMENTED** [review]
code looks good with good practise followed.

File: docs/css/docs.css [hunk 2]
```diff
@@ -120,6 +120,33 @@
   margin-bottom: 0 !important;
```
Comment (lannisterrr) [inline docs/css/docs.css:120]: You should not use important ideally.

File: docs/css/docs.css (hunk 1 above)
Comment (junaidshaikh-js) [reply docs/css/docs.css:125]: This value is used only once in whole documentation website, hence directly used.

File: docs/css/docs.css (hunk 2 above)
Comment (junaidshaikh-js) [reply docs/css/docs.css:120]: It was necessary to use the `!important` here to overwrite  CSS injected by JavaScript in Github Gist HTML

File: docs/css/docs.css (hunk 1 above)
Comment (devananth) [reply docs/css/docs.css:125]: Okay !

**Review by devananth — APPROVED** [review]

File: docs/css/docs.css (hunk 2 above)
Comment (lannisterrr) [reply docs/css/docs.css:120]: cool.

**Review by lannisterrr — APPROVED** [review]

**Pull request merged**